modules/services/window-managers/i3-sway/lib/options.nix

```diff
@@ -69,7 +91,10 @@ let
           '';
         });
     in {
-      fonts = fonts // optionalAttrs versionAtLeast2009 { default = [ ]; };
+      font = mkOption {
```

Comment:
would be nice if we could reuse types.fontType

Reply:
The problem is that this requires a list of fonts, not a single font.

Reply:
wont a `listOf fontType` work ?
https://i3wm.org/docs/userguide.html#fonts

Reply:
The issue is that there can only be a single font size and style specified (that was the original issue described in #1937) for all of the font names.

Reply:
maybe add a comment about fractional scaling, I think it may be surprising for people (it was for me).
Also this is a breaking change why rename the options ? `fonts` was fine, I get `The option `home-manager.users.teto.xsession.windowManager.i3.config.fonts' does not exist.`. In fact I feel like this may be overengineered, we dont need the complexity, we can just fix the example. Even with the change, people can still shoot themselves in the foot. The PR might be useful if someone merges different configurations for the font option but that's really an edgecase I feel like.

Reply:
I agree that people can still shoot themselves in the foot, but I think it will be a lot harder to do so since it is more obvious how you are supposed to use it (as opposed to before where you had to have "insider information" about how both the hm module works and how i3/sway expects fonts to be specified).

I've gone ahead and renamed to `fonts` and added backwards compatibility for the old list-only version of specifying the font.